src/main/java/io/quarkus/gizmo/ClassCreator.java

```diff
@@ -313,9 +313,20 @@ public Builder superClass(Class<?> superClass) {
 
         public Builder setFinal(boolean isFinal) {
             if (isFinal) {
-                extraAccess |= Opcodes.ACC_FINAL;
+                access |= Opcodes.ACC_FINAL;
             } else {
-                extraAccess &= ~Opcodes.ACC_FINAL;
+                access &= ~Opcodes.ACC_FINAL;
+            }
+            return this;
+        }
+
+        public Builder setInterface(boolean isInterface) {
```

Comment:
Do we really need the boolean parameter? I mean, is anyone ever going to call this with `false`?

Reply:
Good point.

Reply:
I think it would be better to have `ClassCreator.interfaceBuilder()` (in addition to `ClassCreator.builder()`), and get rid of `setInterface()` altogether.

Reply:
Sounds good but wouldn't it make sense to introduce `InterfaceCreator` as well? I mean `ClassCreator.Builder.superClass()` methods do not make sense either.

Reply:
Yeah, there's probably a few methods on `ClassCreator.Builder` that don't make sense for interfaces. I've commented on `setFinal`, `superClass` is probably also one of them.

Reply:
I've implemented my ideas here: https://github.com/Ladicek/gizmo/commits/interfaces -- it's this PR's commit amended, so I could just push to @ppalaga's branch if that's OK.

Reply:
Thanks @Ladicek, your proposal looks great! Let me close this one and let's go with a new PR by you.